labels-config-service-impl/src/main/java/org/hypertrace/label/config/service/LabelsConfigServiceImpl.java

```diff
@@ -80,32 +85,41 @@ private static Label buildLabelFromConfig(ConfigObject configObject) {
   @Override
   public void createLabel(
       CreateLabelRequest request, StreamObserver<CreateLabelResponse> responseObserver) {
+    RequestContext requestContext = RequestContext.CURRENT.get();
+    Lock labelsLock = this.stripedLabelsLock.get(requestContext.getTenantId());
     try {
-      RequestContext requestContext = RequestContext.CURRENT.get();
-      CreateLabel createLabel = request.getLabel();
-      if (systemLabelsKeyLabelMap.containsKey(createLabel.getKey())) {
-        // Creating a label with a name that clashes with one of system labels name
-        responseObserver.onError(new StatusRuntimeException(Status.ALREADY_EXISTS));
-        return;
+      boolean lockAcquired = labelsLock.tryLock(5, TimeUnit.MINUTES);
```

Comment:
Please define a static final variable of type Duration, and then use it everywhere. Also `5 seconds` is max you should wait.

Reply:
Done.